- ./libtool is unusable before finishing ./configure
--- libggi-2.2.2/acinclude.m4.orig	2009-10-28 21:27:00.900744957 +0100
+++ libggi-2.2.2/acinclude.m4	2009-10-28 21:27:28.398855073 +0100
@@ -329,7 +329,7 @@
 AC_DEFUN([GGI_CHECK_LIB],
 [
    save_CC="$CC"
-   CC="$SHELL ./libtool --mode=link $CC"
+   CC="$SHELL /usr/bin/libtool --mode=link $CC"
    AC_CHECK_LIB($1, $2, [
      CC="$save_CC"
      $3], [
--- libggi-2.2.2/m4/gii_paths.m4~	2005-09-21 16:38:45.000000000 +0200
+++ libggi-2.2.2/m4/gii_paths.m4	2011-07-25 12:56:39.019829879 +0200
@@ -7,11 +7,9 @@
 [  --with-gii=DIR          use the LibGII installed DIR],
 CFLAGS="$CFLAGS -I$withval/include"
   CPPFLAGS="$CPPFLAGS -I$withval/include"
-  LDFLAGS="$LDFLAGS -L$withval/lib"
   DISTCHECK_CONFIGURE_FLAGS="$DISTCHECK_CONFIGURE_FLAGS --with-gii=$withval",
 CFLAGS="$CFLAGS -I$prefix/include"
   CPPFLAGS="$CPPFLAGS -I$prefix/include"
-  LDFLAGS="$LDFLAGS -L$prefix/lib"
   DISTCHECK_CONFIGURE_FLAGS="$DISTCHECK_CONFIGURE_FLAGS --with-gii=$prefix")
 
 ])
