--- libggi-2.0.1.orig/display/lcd823/mode.c
+++ libggi-2.0.1/display/lcd823/mode.c
@@ -228,16 +228,16 @@
 	}
 	mode->frames = 1;
 
-	if ((tm->dpp.x != 1 && tm->dpp.x != GGI_AUTO) ||
-	    (tm->dpp.y != 1 && tm->dpp.y != GGI_AUTO)) {
+	if ((mode->dpp.x != 1 && mode->dpp.x != GGI_AUTO) ||
+	    (mode->dpp.y != 1 && mode->dpp.y != GGI_AUTO)) {
 		err = -1;
 	}
-	tm->dpp.x = tm->dpp.y = 1;
+	mode->dpp.x = mode->dpp.y = 1;
 
-	if (tm->size.x != GGI_AUTO || tm->size.y != GGI_AUTO) {
+	if (mode->size.x != GGI_AUTO || mode->size.y != GGI_AUTO) {
 		err = -1;
 	}
-	tm->size.x = tm->size.y = GGI_AUTO;
+	mode->size.x = mode->size.y = GGI_AUTO;
 
 	GGIDPRINT_MODE("display-lcd823: result %d %dx%d#%dx%dF%d[0x%02x]\n",
 		       err, mode->visible.x, mode->visible.y,

