--- libggi-2.0.1/configure.in.orig	Sun Aug 26 17:56:45 2001
+++ libggi-2.0.1/configure.in	Sat Apr 27 02:06:26 2002
@@ -451,15 +451,15 @@
 if test "x$build_terminfo_target" != "xno"; then
   curseslib=""
   AC_CHECK_HEADER(ncurses.h,
-	AC_DEFINE(HAVE_NCURSES_H)
-	curseslib="ncurses",
-  AC_CHECK_HEADER(ncurses/ncurses.h,
-	AC_DEFINE(HAVE_NCURSES_NCURSES_H)
-	curseslib="ncurses",
-  AC_CHECK_HEADER(curses.h,
-	AC_DEFINE(HAVE_CURSES_H)
-	curseslib="curses",
-	build_terminfo_target="no")))
+	[AC_DEFINE(HAVE_NCURSES_H)
+	curseslib="ncurses"],
+  [AC_CHECK_HEADER(ncurses/ncurses.h,
+	[AC_DEFINE(HAVE_NCURSES_NCURSES_H)
+	curseslib="ncurses"],
+  [AC_CHECK_HEADER(curses.h,
+	[AC_DEFINE(HAVE_CURSES_H)
+	curseslib="curses"],
+	build_terminfo_target="no")])])
   if test -n "$curseslib"; then
     AC_CHECK_LIB($curseslib, set_term,
 	terminfolibs="-l$curseslib",
